orangecontrib/text/preprocess/tokenize.py

```diff
@@ -0,0 +1,74 @@
+import re
+from nltk import tokenize
+
+
+__all__ = ['BaseTokenizer', 'WordPunctTokenizer', 'PunktSentenceTokenizer',
+           'RegexpTokenizer', 'WhitespaceTokenizer', 'TweetTokenizer']
+
+
+class BaseTokenizer:
+    """ Breaks a string into sequence of tokens. """
+    tokenizer = NotImplemented
+    name = 'Tokenizer'
+
+    def __call__(self, sent):
+        if isinstance(sent, str):
+            return self.tokenize(sent)
+        return self.tokenize_sents(sent)
+
+    def tokenize(self, string):
+        return self.tokenizer.tokenize(string)
+
+    def tokenize_sents(self, strings):
+        return [self.tokenize(string) for string in strings]
+
+    def __str__(self):
+        return self.name
+
+
+class WordPunctTokenizer(BaseTokenizer):
+    tokenizer = tokenize.WordPunctTokenizer()
+    name = 'Word & punctuation'
```

Comment:
The `&` sign is not shown in GUI which makes the name a bit confusing.
![screen shot 2016-06-13 at 16 56 00](https://cloud.githubusercontent.com/assets/713026/16011972/c0be8e38-3187-11e6-8771-3bbb25328419.png)